ITensorGaussianMPS/src/gmps.jl

```diff
@@ -132,15 +79,29 @@ end
 
 function shift!(G::Circuit, i::Int)
   for (n, g) in enumerate(G.rotations)
-    if typeof(g) <: Givens
-      G.rotations[n] = Givens(g.i1 + i, g.i2 + i, g.c, g.s)
-    elseif typeof(g) <: DiagRotation
-      G.rotations[n] = DiagRotation(g.i1 + i, g.i2 + i, g.d)
-    end
+    G.rotations[n] = Givens(g.i1 + i, g.i2 + i, g.c, g.s)
   end
   return G
 end
 
+function scale!(G::Circuit, i::Int)
```

Comment:
Maybe `shift!` and `scale!` could be replaced by some map functions:
```julia
function map!(f, G::Circuit)
  for i in eachindex(G.rotations)
    G.rotations[i] = f(G.rotations[i])
  end
  return G
end
```
and:
```julia
function map_indices!(f, G::Circuit)
  return map!(g -> Givens(f(g.i1), f(g.i2), g.c, g.s)
end
```
Then we can use `map_indices!(i -> i + j, G)` and `map_indices!(i -> i * j, G)` instead of `shift!(G, j)` and `scale!(G, j)` (also `conj!` could be written in terms of `map!`).

Reply:
Sounds good, will take care of it.

Reply:
Actually `replace!` and `replace_indices!` would be better names since that matches the Julia Base definitions (https://docs.julialang.org/en/v1/base/collections/#Base.replace!).